pkg/controllers/user/dnsrecord/dnsrecord.go

```diff
@@ -172,3 +221,30 @@ func (c *EndpointController) reconcileServicesForEndpoint(key string, obj *corev
 
 	return nil
 }
+
+func (c *Controller) queueUpdateForExtNameSvc(key string, delete bool) {
+	alias, ok := serviceUUIDToHostNameAlias.Load(key)
+	if ok && delete {
+		serviceUUIDToHostNameAlias.Delete(key)
+	}
+	if ok {
+		splitted := strings.Split(convert.ToString(alias), "/")
+		namespace := splitted[0]
+		aliasName := splitted[1]
+		c.serviceController.Enqueue(namespace, aliasName)
+	}
+}
+
+func (c *Controller) checkLinkedSvc(namespace string, service string, original string) (bool, string, bool) {
+	svc, err := c.serviceLister.Get(namespace, service)
+	if err != nil && apierrors.IsNotFound(err) {
```

Comment:
what if error != null and the error != not found? Should we not return the error as well?